src/main/java/org/junit/Assert.java

```diff
@@ -1024,13 +1064,27 @@ public static void assertEquals(Object[] expecteds, Object[] actuals) {
                 throw assertionError;
             }
         }
-        String notThrownMessage = buildPrefix(message) + String
-                .format("expected %s to be thrown, but nothing was thrown",
-                        formatClass(expectedThrowable));
+        String notThrownMessage = buildPrefix(message)
+                + String.format("expected %s to be thrown, but nothing was thrown", formatClass(expectedThrowable))
+                + (result == ThrowingSupplierAdapter.NO_VALUE ? "" : String.format(" (returned %s)", result));
```

Comment:
What if the object returned by the lambda throws an exception when you call `toString()` on it?

Reply:
We don't handle that case anywhere, do we? But we should probably call `formatClassAndValue()`. I can make this change but I think we should agree whether to proceed with this PR first.

@kcooney WDYT?

Reply:
Note that `toString()` for arrays ends up being ugly, in case the return value is an array.

I addressed that in JUnit Jupiter like this: https://github.com/junit-team/junit5/commit/c5c4741921480f0f78118aee70a6a6d53405e838

Reply:
> We don't handle that case anywhere, do we?

No, I don't think there is any exception handling for invocations of `toString()` on objects supplied to assertion methods in either JUnit 4 or JUnit Jupiter.